pkg/tree/schema.ipldsch

```diff
@@ -0,0 +1,41 @@
+type ProllyNode struct {
+    # true for leaf nodes and false for branch nodes
+    isLeaf Bool
+    # raw keys(input from users) for leaf node. For branch node, the key is last key in the child node
+	keys [Bytes]
+	# input data from users for leaf nodes, link(cid) for branch nodes
+    values [Any]
+} representation tuple
+
+type ProllyRoot struct{
+    # root ProllyNode of tree
+    RootCid &ProllyNode
+    # config about the tree
+    ConfigCid &TreeConfig
+}
+
+type SuffixThresholdConfig struct {
+    chunkingFactor Int
+} representation tuple
+
+# Tree Config for prolly tree, it includes some global setting, the splitter method you choose and specific configs about
+# the splitter
+type TreeConfig struct{
+    strategyType    Int
+    minNodeSize     Int
+    maxNodeSize     Int
+    maxPairsInNode  Int
+    nodeCodec       Prefix
+    strategy        strategy
+} representation tuple
+
+type strategy union {
+    | SuffixThresholdConfig     "suffix"
+} representation keyed
+
+type Prefix struct{
```

Comment:
Where is this being used?

Reply:
I extract the fields below in ProllyTreeConfig as the Prefix struct, they are used in TreeNode saving(in NodeStore and get cid), we should put these fields in ProllyTreeConfig directly?
  cidVersion Int
  codec Int
  hashFunction Int
  nullable hashLength Int

Reply:
In the spec we had it in the config itself. I think it'd be useful to have it all in one place